backend/repository/thrift_update.go

```diff
@@ -144,3 +145,72 @@ func (r *Repository) ThriftFileVersion(thriftFile string) (string, error) {
 	}
 	return thriftMeta.Version, nil
 }
+
+// DeleteThriftFile deletes specified thrift from disk, including any empty parent directories, and remove its entry from thrift meta
+func (r *Repository) DeleteThriftFile(thriftFile string) error {
+	cfg, err := r.LatestGatewayConfig()
+	if err != nil {
+		return errors.Wrap(err, "invalid configuration before updating thrifts")
+	}
+
+	// Check client and endpoint dependencies
+	for _, client := range cfg.Clients {
+		if client.ThriftFile == thriftFile {
+			return errors.Errorf("Cannot delete thrift %s while client %s is associated", thriftFile, client.Name)
+		}
+	}
+	for _, endpoint := range cfg.Endpoints {
+		if endpoint.ThriftFile == thriftFile {
+			return errors.Errorf("Cannot delete thrift %s while endpoint %s.%s is associated", thriftFile, endpoint.ID, endpoint.HandleID)
+		}
+	}
+
+	meta, err := r.ThriftConfig(cfg.ThriftRootDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to read current thrift meta")
+	}
+
+	// Delete entry	in meta json and write to file
+	delete(meta, thriftFile)
```

Comment:
will there be an issue if one client opens a diff to delete thrift (ok) while another client opens a diff to add client based on the thrift? That two diffs open at the same time.

Reply:
I guess that depends if once the 1st diff is landed, will the 2nd diff get re-built based on the new master? If so, the build should fail right